### PR TITLE
chore: add e2e test, refactor/lint, upgrade deps

### PR DESCRIPTION
**Pull request by Bkucera**
add e2e test
- adds e2e test that snapshots cypress stdout

refactor/ lint
- rename tests/snapshots
- lint files
- add yarn.lock

upgrade deps
- upgrade eslint and use new eslint deps

- [x] use cypress docker image in CI

### CODE REVIEW

File: test/e2e/helpers.js [hunk 1]
```diff
@@ -0,0 +1,379 @@
+/* eslint-disable no-console, prefer-rest-params */
```
Comment (Bkucera) [inline test/e2e/helpers.js]: i know i know, this file's a little rough

Comment (Bkucera) [reply test/e2e/helpers.js]: i use a similar approach in the retries plugin

File: circle.yml [hunk 2]
```diff
@@ -9,19 +9,27 @@ workflows:
...
   test:
     docker:
-      - image: circleci/node:8
+      - image: cypress/base:8
 
     steps:
       - checkout
-
+      
+      - restore_cache:
```
Comment (bahmutov) [inline circle.yml]: I suggest we start using https://circleci.com/orbs/registry/orb/circleci/node because it simplifies installing and caching a lot

Comment (bahmutov) [reply circle.yml]: or even our Cypress orb

**Pull request merged**